zookeeper/core/cli_test.py

```diff
@@ -42,11 +42,24 @@ def test_pass_param_values(test_task_runner):
     assert result.output == "5 foo False\n"
 
 
-def test_param_key_valid_characters(test_task_runner):
+def test_param_key_valid_characters():
```

Comment:
This test had to be modified because it previously relied on the behaviour that this PR is explicitly disallowing.